adlik_serving/runtime/enflame/model/enflame_model.cc

```diff
@@ -0,0 +1,298 @@
+// Copyright 2019 ZTE corporation. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "adlik_serving/runtime/enflame/model/enflame_model.h"
+
+#include <algorithm>
+#include <cstring>
+#include <numeric>
+
+#include <TopsInference/TopsInferRuntime.h>
+#include <dtu/factor/program.h>
+#include "adlik_serving/framework/domain/model_config_helper.h"
+#include "adlik_serving/runtime/enflame/model/enflame_util.h"
+#include "adlik_serving/runtime/provider/predict_request_provider.h"
+#include "adlik_serving/runtime/provider/predict_response_provider.h"
+#include "adlik_serving/runtime/util/datatype_size.h"
+#include "cub/env/fs/path.h"
+#include "cub/log/log.h"
+#include "tensorflow/core/lib/io/path.h"
+#include "tensorflow/core/platform/env.h"
+#include "tensorflow/core/platform/errors.h"
+
+namespace enflame_runtime {
+
+using namespace adlik::serving;
+using namespace TopsInference;
+
+namespace {
+
+#define MY_RETURN_IF_ERROR(...)                         \
+  do {                                                  \
+    const ::tensorflow::Status _status = (__VA_ARGS__); \
+    if (TF_PREDICT_FALSE(!_status.ok()))                \
+      return cub::Status(_status.code());               \
+  } while (0)
+
+#define MY_RETURN_IF_ERROR_TF(...)                      \
+  do {                                                  \
+    const ::tensorflow::Status _status = (__VA_ARGS__); \
+    if (TF_PREDICT_FALSE(!_status.ok()))                \
+      return _status;                                   \
+  } while (0)
+
+struct Buffer {
+  Buffer(std::string name, tensorflow::DataType data_type, void* address = nullptr)
+      : name(name), data_type(data_type), address(address) {
+  }
+  ~Buffer() {
+    TopsInference::mem_free(address);
+  }
+  std::string name;
+  tensorflow::DataType data_type;
+  size_t byte_1batch_size;
+  size_t byte_size;
+  std::vector<int> shape;
+  void* address = nullptr;
+};
+
+struct Instance : BatchProcessor {
+  Instance(const std::string& name, const adlik::serving::ModelConfigProto& config, TopsInference::IEngine* model)
+      : name(name), config(config), engine(model), stream(stream), max_batch_size(config.max_batch_size()) {
```

Comment:
1. Where is the input parameter "stream"?
2. The initializer list should be match in member declaration order